packages/bundler-vite/src/resolveViteConfig.ts

```diff
@@ -15,16 +15,18 @@ export const resolveViteConfig = async ({
   options: ViteBundlerOptions
   isBuild: boolean
   isServer: boolean
-}): Promise<InlineConfig> =>
-  mergeConfig(
+}): Promise<InlineConfig> => {
+  const viteOptions = options.viteOptions ?? {}
+  return mergeConfig(
     {
       clearScreen: false,
       configFile: false,
       logLevel: !isBuild || app.env.isDebug ? 'info' : 'warn',
       plugins: [
         vuePlugin(options.vuePluginOptions),
-        vuepressPlugin({ app, isBuild, isServer }),
+        vuepressPlugin({ app, isBuild, isServer, viteOptions }),
       ],
     },
-    options.viteOptions ?? {}
+    viteOptions
```

Comment:
Why should I use a plugin for this?
Choosing a minifier is part of Vite configuration:
https://vitejs.dev/config/build-options.html#build-minify

Reply:
If so, user could choose a minifier via `viteOptions` directly right?

But this PR means that `viteOptions` won't take effects, because the minifier option from plugins overrides it, doesn't it?

Reply:
Current `minify` value is `true` if it is not in  development mode:
https://github.com/vuepress/vuepress-next/blob/main/packages/bundler-vite/src/plugins/mainPlugin.ts#L93

Changes in the pull request assume that the `minify` value is taken from the users `viteOptions` in VuePress configuration file or is set to true. But in development mode the value is always false.

Reply:
Well, I might not express myself clearly.

- Currently, users `viteOptions` won't take effect, right?
- Why? The `viteOptions` is merged into vite config after our `mainPlugin`, right?
- So maybe the reason is that, those config from plugin has higher priority than `viteOptions` (I'm not sure, though, have not tested)
- If so, maybe we could create another plugin with `viteOptions`, and push the new plugin after `vuePlugin` and `vuepressPlugin`, then the `viteOptions` might have the highest priority.

Anyway, it's just an assumption, and need to be verified.

Reply:
Here is a part of my VuePress plugin:

```ts
  bundler: viteBundler({
    viteOptions: {
      build: {
        minify: 'terser',
        terserOptions: {
          format: {
            comments: false,
          },
        },
      },
    }
  })
```

When I build app I get the following error:

```
Compiling with vite
build.terserOptions is specified but build.minify is not set to use Terser. Note Vite now defaults to use esbuild for minification. If you still prefer Terser, set build.minify to "terser".
```

Reply:
Thus, maybe we could simply using following snippet here to allow user's `viteOptions` always take effects.

```ts
      plugins: [
        vuePlugin(options.vuePluginOptions),
        vuepressPlugin({ app, isBuild, isServer }),
        { config: () => options.viteOptions ?? {} },
      ],
```